python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/shape_lib_gen.py

```diff
@@ -568,6 +568,19 @@ def _reduce_along_dim(self: List[int], dim: int, keepdim: bool):
             out.append(self_dim)
     return out
 
+def _reduce_along_dims(self: List[int], dims: List[int], keepdim: bool):
+    wrapped_dims: List[int] = []
+    for dim in dims:
+        wrapped_dims.append(upstream_shape_functions.maybe_wrap_dim(dim, len(self)))
+    out: List[int] = []
+    for i, self_dim in enumerate(self):
+        if i in wrapped_dims:
+            if keepdim:
+                out.append(1)
+        else:
+            out.append(self_dim)
+    return out
+
```

Comment:
You have to upstream this in the PyTorch. Like https://github.com/pytorch/pytorch/pull/80860.

Reply:
Yes, I'll keep this in mind and work on adding this in PyTorch along with `aten.upsample_nearest2d`'s shape function as mentioned by @ramiro050 [here](https://github.com/llvm/torch-mlir/pull/1590#discussion_r1025437892).